### PR TITLE
Fix parsing issues in Firestore node

### LINKED ISSUE #1718: Firestore saves strings with number at the end as timestamp and not as string
**Describe the bug**
When a string value with a number at the end is saved with Google Firebase - Cloud Firestore the number is saved as a timestamp instead of string.

**To Reproduce**

<details>
<summary>Test Workflow</summary>

```json
{
  "name": "Test",
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        240,
        360
      ]
    },
    {
      "parameters": {
        "operation": "upsert",
        "projectId": "testproject",
        "collection": "tests",
        "updateKey": "id",
        "columns": "name"
      },
      "name": "Testsave",
      "type": "n8n-nodes-base.googleFirebaseCloudFirestore",
      "typeVersion": 1,
      "position": [
        690,
        500
      ],
      "alwaysOutputData": true,
      "retryOnFail": true,
      "credentials": {
        "googleFirebaseCloudFirestoreOAuth2Api": "Firestore"
      }
    },
    {
      "parameters": {
        "functionCode": "return [\n  {\n    json: {\n      id: 1,\n      name: \"TEST SM UPDATED 2\"\n    },\n  },\n  {\n    json: {\n      id: 2,\n      name: \"ABC 2\"\n    },\n  },\n  {\n    json: {\n      id: 3,\n      name: \"blabla 6\"\n    },\n  },\n];"
      },
      "name": "Function",
      "type": "n8n-nodes-base.function",
      "typeVersion": 1,
      "position": [
        500,
        500
      ]
    }
  ],
  "connections": {
    "Function": {
      "main": [
        [
          {
            "node": "Testsave",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Start": {
      "main": [
        []
      ]
    }
  },
  "active": false,
  "settings": {
    "errorWorkflow": "2",
    "saveDataSuccessExecution": "none",
    "saveDataErrorExecution": "none",
    "executionTimeout": -1
  },
  "id": "6"
}

```

</details>

E.g. These strings:

```javascript
const values = [
    "TEST SM UPDATED 2",
    "ABC 2",
    "blabla 6",
];
```

Are saved as:
- February 1, 2001 at 1:00:00 AM UTC+1 (timestamp)
- February 1, 2001 at 1:00:00 AM UTC+1 (timestamp)
- June 1, 2001 at 2:00:00 AM UTC+2 (timestamp)

**Expected behavior**
That the strings are saved as strings and not timestamps.

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04
 - n8n Version 0.117.0
 - Node.js Version ?? (I use the docker image)

**Additional context**
I checked if it is an error with the [jsonToDocument](https://github.com/n8n-io/n8n/blob/n8n%400.117.0/packages/nodes-base/nodes/Google/Firebase/CloudFirestore/GenericFunctions.ts#L65) function in [this playground](https://www.typescriptlang.org/play?ssl=12&ssc=37&pln=12&pc=38#code/GYVwdgxgLglg9mABAKwM4ICpwCJwiAWwFMwoAKANwEMAbEIgLkVSgCcYwBzRAH0TEIAjIq0QBKJlTABPRAG8AUAEgYwRJVr1EAXl2IA5G3r7efanSI69+4LVRETPPlGkAHInDXmtu7QcFwcDREUvpi8spKrERQIKxIcv6BwVIAapoOTN6WAL4A3Mo5iEQ09oiq6tlWfgI0NOGKSlExcQkGtTTpFvpMHYj5hcWllhVkAIQwqAByVFMaFohUqPxCImINkaPZAHRQcADKbBycZGLbHAAmRAAeAPLAZPrbYYhjegC0AIwbTc2x8fIDBc4CBBMEusYshl+gUmkUSmVGk1ov82voOFAiJwRBDMogqgM4YMESM1GRsFRMdtXFRWPZ5lolswjlx1hEmhAECxEBdKZYakQAO6ICmY8l86m0+lVJksdissSwv6tQGGGDEFhUAiuXE9HkSvYASX2t0O8pO4UJ8OG5TJLncnnx0N8BjlxxeSJRKsS+jdXF1UIWVqGZS20IAZOGnRZtpywHKQNA4KIXQBBVisKjSDaIXN5vMAegLAAEoKh3jBOGBk0RIl6AT7aZnpAHAdlUIH6NsCFRXJQdAA+FDoMBYXD4YikShsorBkm29T2jxeZ3WOCCZBEaAeyJx7nr5A6eSEpTAZPqPdQRBwa9qW4brdQbYAayI0lQDKIbKRSiLpfLlbVtEkRKAeADacAALpHmgmA4HghAkOQ2QQZBiqRDkdYtA2Bg9jqGR6okwAwCUFwdteG79DChQKMo9YJAMmEKJe0b0MsfhgQo+YAEQYAAovsGCIPsACyiAAKoAArYKm-HYIgABM3EADRcXm3GpgAQgAwopKlqbm3FglQxmIAAbPpkEFAo7bbGerB8VQEAABZkKAkCwAg6gwJiBDKeUYBXNcOaIHuQRENsNBwCcPlEH5w5weOiFTrFBDrAUOToUAA) but the values are recognized correctly as `stringValue`.

### PR DESCRIPTION
**Pull request by ivov**
- Fix parsing for [empty document](https://github.com/n8n-io/n8n/pull/1749#discussion_r628238125)
- Fix parsing for [geopoint](https://github.com/n8n-io/n8n/pull/1749#discussion_r628243690)
- Fix detection for timestamp to close #1718

Regarding the third fix:

Both `Date.parse(str)` and `moment(str).isValid()` return a date for `"TEST SM UPDATED 2"`, `"TEST SM UPDATED-2"` and `"TEST SM UPDATED/2"`. Same with standalone strings `"1"` to `"12"`, and `"32"` and above. To prevent date misdetection, the parser now strictly requires a specific format with `moment(str, 'YYYY-MM-DD HH:mm:ss Z', true)` to create a Firestore timestamp value. See [Moment.js time tokens reference](https://momentjs.com/docs/#/parsing/string-format/).

### CODE REVIEW

File: packages/nodes-base/nodes/Google/Firebase/CloudFirestore/GenericFunctions.ts [hunk 1]
```diff
@@ -58,6 +60,7 @@ export async function googleApiRequestAllItems(this: IExecuteFunctions | ILoadOp
 	return returnData;
 }
 
+const isValidDate = (str: string) => moment(str, 'YYYY-MM-DD HH:mm:ss Z', true).isValid();
```
Comment (MedAliMarz) [inline packages/nodes-base/nodes/Google/Firebase/CloudFirestore/GenericFunctions.ts:63]: Maybe adding **ISO_8601** format can make it easy for users to provide an ISO datetime as input.
```js
 const isValidDate = (str: string) => moment(str, ['YYYY-MM-DD HH:mm:ss Z', moment.ISO_8601], true).isValid();

```

**Review by MedAliMarz — COMMENTED** [review]
The issues are solved, Thanks @ivov.

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.120.0

**Pull request merged**